pandas/core/tools/datetimes.py

```diff
@@ -103,6 +106,70 @@ def _convert_and_box_cache(arg, cache_array, box, errors, name=None):
     return result.values
 
 
+def _return_parsed_timezone_results(result, tznames, tzoffsets, parsing_tzname,
+                                    parsing_tzoffset, box):
+    """
+    Return results from array_strptime if a %z or %Z directive was passed.
+
+    If %Z is only parsed, timezones will be a pytz.timezone object.
```

Comment:
why are you returning a datetime.timezone at all? we barely use these, prefer pytz.timezone

how is it possible %z and %Z both are parsed?